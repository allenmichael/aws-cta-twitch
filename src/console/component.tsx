import * as React from 'react';
import './component.sass';

interface ConsoleEntry {
  frame: string;
  log: string;
}

interface State {
  logHistory: ConsoleEntry[];
}

export class Console extends React.Component<{}, State> {
  public state: State = {
    logHistory: window.rig.history || []
  }

  private updateConsole() {
    this.setState({
      logHistory: window.rig.history
    });
  }

  public componentDidMount() {
    window.rig.update = () => {
      this.updateConsole();
    }

    this.updateConsole();
  }

  public render() {
    const logs = this.state.logHistory.map((entry, index) => (
      <div key={index}>{entry.frame} $ {entry.log}</div>
    ));

    return (
      <div className="console">{logs}</div>
    );
  }
}
