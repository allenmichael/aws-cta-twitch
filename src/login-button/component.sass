.login-button
  float: right
  color: white
  background-color: #7D55C7
  border: 1px solid white
  text-align: center
  width: 5rem
  font-size: 13px
  vertical-align: middle
  line-height: 25px
  cursor: pointer
