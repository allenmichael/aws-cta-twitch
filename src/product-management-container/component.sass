.product-container-wrapper
  overflow-y: auto

.product-container
  display: flex
  flex-direction: row
  margin: 1em
  flex-wrap: wrap
  justify-content: center
  align-items: stretch