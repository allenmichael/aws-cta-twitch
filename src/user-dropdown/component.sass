.user-dropdown
  display: flex
  flex-direction: column
  align-items: center
  font-size: 14px
  color: white
  background-color: #7D55C7
  text-align: center
  vertical-align: middle
  height: 3rem

  &:focus
    background-color: #4b367c
    outline-color: #c5b6e2
    outline-width: 0px

.user-dropdown__name-container
  width: 10rem
  margin-top: 0.25rem
  display: flex
  flex-direction: row
  align-items: center
  justify-content: space-evenly

.user-dropdown__image
  width: 30px
  height: 30px
  margin: 0.3rem 0.3rem

.user-dropdown__triangle
  color: #fff
  height: 10px
  width: 10px
  transform: rotate(180deg)
  -webkit-transition: transform 0.3s ease
  -moz-transition: transform 0.3s ease
  -o-transition: transform 0.3s ease
  -ms-transition: transform 0.3s ease
  transition: transform 0.3s ease

.user-dropdown__triangle.open
  transform: rotate(0deg)

.user-dropdown__menu
  position: absolute
  text-align: left
  margin-top: 3rem
  font-size: 12px
  color: white
  background-color: rgba(32, 29, 44, 0.8)
  width: 10rem
  height: 0
  visibility: hidden
  -webkit-transition: height 0.3s ease
  -moz-transition: height 0.3s ease
  -o-transition: height 0.3s ease
  -ms-transition: height 0.3s ease
  transition: height 0.3s ease, visibility 0.3s
  z-index: 1000
  >ul
    >li
      margin-top: 0.5rem
      >a
        color: white
        text-decoration: none
        >img
          margin-left: .5em

    list-style: none
    margin: 1rem
    padding: 0

.user-dropdown__menu.open
  height: 9rem
  visibility: visible

.user-dropdown__divider
  width: 100%
  border-top: 1px solid grey
