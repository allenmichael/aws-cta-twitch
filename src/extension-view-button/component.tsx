import * as React from 'react';
import * as line from '../img/rig_line.png';
import * as plus from '../img/plus_icon.png';
import './component.sass'

interface ExtensionViewButtonProps {
  onClick: () => void;
}

type Props = ExtensionViewButtonProps;
export class ExtensionViewButton extends React.Component<Props> {
  public render() {
    return (
      <div className='button-wrapper'>
        <img alt='line' className='background' src={line} />
        <div className="circle-button" onClick={this.props.onClick}>
          <img alt='Add new view' src={plus} />
        </div>
      </div>
    )
  }
}
