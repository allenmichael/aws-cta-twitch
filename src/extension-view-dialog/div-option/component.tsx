import * as React from 'react';
import './component.sass';
import './component.css';

interface DivOptionProps {
  img: string;
  name: string;
  value: string;
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
  checked: boolean;
}
export class DivOption extends React.Component<DivOptionProps> {
  public render() {
    return (
      <div className="extension-container__block">
        <label className="extension-type-label">
          <input type="radio" name="extensionViewType" value={this.props.value} onChange={this.props.onChange} defaultChecked={this.props.checked}/>
          <div className="extension-type-box">
            <img
              className='extension-type-box__img'
              alt={this.props.value}
              src={this.props.img} />
          </div>
          <div className={this.props.checked ? 'extension-type-text dialog-selected': 'extension-type-text'}>{this.props.name}</div>
        </label>
      </div>
    );
  }
}
