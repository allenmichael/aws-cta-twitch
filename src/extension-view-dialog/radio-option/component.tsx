import * as React from 'react';
import './component.sass';

interface RadioOptionProps {
  name: string;
  value: string;
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
  checked: boolean;
}
type Props = RadioOptionProps & React.HTMLAttributes<HTMLInputElement>
export class RadioOption extends React.Component<Props> {
  public render() {
    return (
      <div className='option-div'>
        <label className="option-label">
            <input type="radio" name={this.props.name} value={this.props.value} onChange={this.props.onChange} defaultChecked={this.props.checked}/>
            <span className={this.props.checked ? 'dialog-selected': null}>{this.props.value}</span>
        </label>
      </div>
    )
  }
}
