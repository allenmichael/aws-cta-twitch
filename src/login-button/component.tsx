import * as React from 'react';
import './component.sass';

export class LoginButton extends React.Component<{}>{
  public render() {
    const rigAuthUrl = 'https://id.twitch.tv/oauth2/authorize?client_id=rtwj27xsv591q1zcnqqfg7suduj4pq&redirect_uri=https://localhost.rig.twitch.tv:3000&response_type=token&scope=user:edit+extensions:edit:catalog';
    return (
        <a href={rigAuthUrl}>
          <button className='login-button'>Sign In</button>
        </a>
    );
  }
}
