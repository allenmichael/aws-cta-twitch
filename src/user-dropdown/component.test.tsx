import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { UserDropdownComponent } from './component';

describe('<UserDropdownComponent />', () => {
  const defaultGenerator = () => ({
    session: { login: 'test', profileImageUrl: 'test.png', authToken: 'test' },
  });
  const setupRenderer = setupShallowTest(UserDropdownComponent, defaultGenerator);

  it('renders correctly', () => {
    const { wrapper } = setupRenderer();
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render if session null', () => {
    const { wrapper } = setupRenderer({
      session: null,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('remders correctly when clicked', () => {
    const { wrapper } = setupRenderer();
    wrapper.simulate('click');
    expect(wrapper.find('.user-dropdown__menu.open')).toHaveLength(1);

    wrapper.simulate('click');
    expect(wrapper.find('.open')).toHaveLength(0);
  });
});
