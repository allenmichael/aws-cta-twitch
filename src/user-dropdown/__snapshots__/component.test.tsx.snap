// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDropdownComponent /> does not render if session null 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDropdownComponent
    session={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<UserDropdownComponent /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDropdownComponent
    session={
        Object {
            "authToken": "test",
            "login": "test",
            "profileImageUrl": "test.png",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-dropdown__name-container"
>
          <img
                    alt="test"
                    className="user-dropdown__image"
                    src="test.png"
          />
          <div
                    className="user-dropdown__username"
          >
                    test
          </div>
          <img
                    className="user-dropdown__triangle"
                    src="whitetriangle.svg"
          />
</div>,
        <div
          className="user-dropdown__menu transition"
>
          <ul>
                    <li>
                              <a
                                        href="https://dev.twitch.tv/dashboard/extensions"
                                        target="_blank"
                              >
                                        Extensions Dashboard
                              </a>
                    </li>
                    <li>
                              <a
                                        href="https://dev.twitch.tv/docs/extensions/"
                                        target="_blank"
                              >
                                        Documentation
                              </a>
                    </li>
                    <li>
                              <a
                                        href="mailto:developers@twitch.tv"
                              >
                                        Give Feedback
                              </a>
                    </li>
                    <li>
                              <a
                                        href={null}
                                        target="_blank"
                              >
                                        Rig Up To Date
                              </a>
                    </li>
                    <li>
                              <div
                                        className="user-dropdown__divider"
                              />
                    </li>
                    <li
                              onClick={[Function]}
                    >
                              Sign Out
                    </li>
          </ul>
</div>,
      ],
      "className": "user-dropdown",
      "onClick": [Function],
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <img
              alt="test"
              className="user-dropdown__image"
              src="test.png"
/>,
            <div
              className="user-dropdown__username"
>
              test
</div>,
            <img
              className="user-dropdown__triangle"
              src="whitetriangle.svg"
/>,
          ],
          "className": "user-dropdown__name-container",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "test",
              "className": "user-dropdown__image",
              "src": "test.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test",
              "className": "user-dropdown__username",
            },
            "ref": null,
            "rendered": "test",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user-dropdown__triangle",
              "src": "whitetriangle.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul>
            <li>
                        <a
                                    href="https://dev.twitch.tv/dashboard/extensions"
                                    target="_blank"
                        >
                                    Extensions Dashboard
                        </a>
            </li>
            <li>
                        <a
                                    href="https://dev.twitch.tv/docs/extensions/"
                                    target="_blank"
                        >
                                    Documentation
                        </a>
            </li>
            <li>
                        <a
                                    href="mailto:developers@twitch.tv"
                        >
                                    Give Feedback
                        </a>
            </li>
            <li>
                        <a
                                    href={null}
                                    target="_blank"
                        >
                                    Rig Up To Date
                        </a>
            </li>
            <li>
                        <div
                                    className="user-dropdown__divider"
                        />
            </li>
            <li
                        onClick={[Function]}
            >
                        Sign Out
            </li>
</ul>,
          "className": "user-dropdown__menu transition",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <a
                                href="https://dev.twitch.tv/dashboard/extensions"
                                target="_blank"
                >
                                Extensions Dashboard
                </a>
</li>,
              <li>
                <a
                                href="https://dev.twitch.tv/docs/extensions/"
                                target="_blank"
                >
                                Documentation
                </a>
</li>,
              <li>
                <a
                                href="mailto:developers@twitch.tv"
                >
                                Give Feedback
                </a>
</li>,
              <li>
                <a
                                href={null}
                                target="_blank"
                >
                                Rig Up To Date
                </a>
</li>,
              <li>
                <div
                                className="user-dropdown__divider"
                />
</li>,
              <li
                onClick={[Function]}
>
                Sign Out
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="https://dev.twitch.tv/dashboard/extensions"
                  target="_blank"
>
                  Extensions Dashboard
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Extensions Dashboard",
                  "href": "https://dev.twitch.tv/dashboard/extensions",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Extensions Dashboard",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="https://dev.twitch.tv/docs/extensions/"
                  target="_blank"
>
                  Documentation
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Documentation",
                  "href": "https://dev.twitch.tv/docs/extensions/",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Documentation",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="mailto:developers@twitch.tv"
>
                  Give Feedback
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Give Feedback",
                  "href": "mailto:developers@twitch.tv",
                },
                "ref": null,
                "rendered": "Give Feedback",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href={null}
                  target="_blank"
>
                  Rig Up To Date
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Rig Up To Date",
                    false,
                  ],
                  "href": null,
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Array [
                  "Rig Up To Date",
                  false,
                ],
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="user-dropdown__divider"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "user-dropdown__divider",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Out",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-dropdown__name-container"
>
            <img
                        alt="test"
                        className="user-dropdown__image"
                        src="test.png"
            />
            <div
                        className="user-dropdown__username"
            >
                        test
            </div>
            <img
                        className="user-dropdown__triangle"
                        src="whitetriangle.svg"
            />
</div>,
          <div
            className="user-dropdown__menu transition"
>
            <ul>
                        <li>
                                    <a
                                                href="https://dev.twitch.tv/dashboard/extensions"
                                                target="_blank"
                                    >
                                                Extensions Dashboard
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="https://dev.twitch.tv/docs/extensions/"
                                                target="_blank"
                                    >
                                                Documentation
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="mailto:developers@twitch.tv"
                                    >
                                                Give Feedback
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href={null}
                                                target="_blank"
                                    >
                                                Rig Up To Date
                                    </a>
                        </li>
                        <li>
                                    <div
                                                className="user-dropdown__divider"
                                    />
                        </li>
                        <li
                                    onClick={[Function]}
                        >
                                    Sign Out
                        </li>
            </ul>
</div>,
        ],
        "className": "user-dropdown",
        "onClick": [Function],
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <img
                alt="test"
                className="user-dropdown__image"
                src="test.png"
/>,
              <div
                className="user-dropdown__username"
>
                test
</div>,
              <img
                className="user-dropdown__triangle"
                src="whitetriangle.svg"
/>,
            ],
            "className": "user-dropdown__name-container",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "test",
                "className": "user-dropdown__image",
                "src": "test.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
                "className": "user-dropdown__username",
              },
              "ref": null,
              "rendered": "test",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "user-dropdown__triangle",
                "src": "whitetriangle.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul>
              <li>
                            <a
                                          href="https://dev.twitch.tv/dashboard/extensions"
                                          target="_blank"
                            >
                                          Extensions Dashboard
                            </a>
              </li>
              <li>
                            <a
                                          href="https://dev.twitch.tv/docs/extensions/"
                                          target="_blank"
                            >
                                          Documentation
                            </a>
              </li>
              <li>
                            <a
                                          href="mailto:developers@twitch.tv"
                            >
                                          Give Feedback
                            </a>
              </li>
              <li>
                            <a
                                          href={null}
                                          target="_blank"
                            >
                                          Rig Up To Date
                            </a>
              </li>
              <li>
                            <div
                                          className="user-dropdown__divider"
                            />
              </li>
              <li
                            onClick={[Function]}
              >
                            Sign Out
              </li>
</ul>,
            "className": "user-dropdown__menu transition",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <a
                                    href="https://dev.twitch.tv/dashboard/extensions"
                                    target="_blank"
                  >
                                    Extensions Dashboard
                  </a>
</li>,
                <li>
                  <a
                                    href="https://dev.twitch.tv/docs/extensions/"
                                    target="_blank"
                  >
                                    Documentation
                  </a>
</li>,
                <li>
                  <a
                                    href="mailto:developers@twitch.tv"
                  >
                                    Give Feedback
                  </a>
</li>,
                <li>
                  <a
                                    href={null}
                                    target="_blank"
                  >
                                    Rig Up To Date
                  </a>
</li>,
                <li>
                  <div
                                    className="user-dropdown__divider"
                  />
</li>,
                <li
                  onClick={[Function]}
>
                  Sign Out
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="https://dev.twitch.tv/dashboard/extensions"
                    target="_blank"
>
                    Extensions Dashboard
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Extensions Dashboard",
                    "href": "https://dev.twitch.tv/dashboard/extensions",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Extensions Dashboard",
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="https://dev.twitch.tv/docs/extensions/"
                    target="_blank"
>
                    Documentation
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Documentation",
                    "href": "https://dev.twitch.tv/docs/extensions/",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Documentation",
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="mailto:developers@twitch.tv"
>
                    Give Feedback
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Give Feedback",
                    "href": "mailto:developers@twitch.tv",
                  },
                  "ref": null,
                  "rendered": "Give Feedback",
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href={null}
                    target="_blank"
>
                    Rig Up To Date
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Rig Up To Date",
                      false,
                    ],
                    "href": null,
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Rig Up To Date",
                    false,
                  ],
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="user-dropdown__divider"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "user-dropdown__divider",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Out",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign Out",
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
